### PR TITLE
Azure Event Grid API implementation

### PR DESCRIPTION
**Pull request by sjuarezgx**
Issue:103791

### CODE REVIEW

File: dotnet/src/dotnetcore/Providers/Messaging/GXAzureEventGrid/AzureEventGrid.cs [hunk 1]
```diff
@@ -0,0 +1,338 @@
...
+		}
+		public AzureEventGrid(GXService providerService) : base(providerService)
+		{
+			Initialize(providerService);
+		}
+		private void Initialize(GXService providerService)
+		{
+			ServiceSettings serviceSettings = new(PropertyConstants.EVENT_ROUTER, Name, providerService);
+			_endpoint = serviceSettings.GetEncryptedPropertyValue(PropertyConstants.URI_ENDPOINT);
+			_accessKey = serviceSettings.GetEncryptedPropertyValue(PropertyConstants.ACCESS_KEY);
```
Comment (ggallotti) [inline dotnet/src/dotnetcore/Providers/Messaging/GXAzureEventGrid/AzureEventGrid.cs:33]: Is this the recommended security approach to connect with Azure Services?

In AWS, Companies such as Globant, does not allow to connect to Any service using Security Credentials. They must use IAM Roles to access sensitive services. Is there anything equivalent in Azure?

Comment (ggallotti) [reply dotnet/src/dotnetcore/Providers/Messaging/GXAzureEventGrid/AzureEventGrid.cs:33]: Maybe DefaultAzureCredential ?

File: dotnet/src/dotnetcore/Providers/Messaging/GXAzureEventGrid/AzureEventGrid.cs [hunk 2]
```diff
@@ -0,0 +1,338 @@
...
+
+			//Package Azure.Identity cannot be referenced because it conflicts with Microsoft.Identity.Client pack version
+			//used transitively by gxmail and dynamoDB projects.
+
+			/*if (string.IsNullOrEmpty(_accessKey))
+
+				//Try using Active Directory authentication
```
Comment (ggallotti) [inline dotnet/src/dotnetcore/Providers/Messaging/GXAzureEventGrid/AzureEventGrid.cs:40]: Can we remove this commented code?

File: dotnet/src/dotnetcore/Providers/Messaging/GXAzureEventGrid/EventGridRouterProvider.cs [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+				{ PropertyConstants.EVENTROUTER_AZUREEG_ENDPOINT, endpoint },
+				{ PropertyConstants.EVENTROUTER_AZUREEG_ACCESS_KEY, accesskey }
+			};
+
+			EventRouterProviderBase evtRouterProvider = eventRouterProvider.Connect(PropertyConstants.AZUREEVENTGRID, properties, out GXBaseCollection<SdtMessages_Message> errorMessagesConnect, out bool successConnect);
+			errorMessages = errorMessagesConnect;
+			success = successConnect;
+			return evtRouterProvider;
+		}
+		/*public EventRouterProviderBase Connect(string endpoint, out GXBaseCollection<SdtMessages_Message> errorMessages, out bool success)
```
[the remark this answers is not in the document]
Comment (sjuarezgx) [reply dotnet/src/dotnetcore/Providers/Messaging/GXAzureEventGrid/EventGridRouterProvider.cs]: ready

File: dotnet/src/dotnetcore/Providers/Messaging/GXAzureEventGrid/AzureEventGrid.cs (hunk 2 above)
Comment (sjuarezgx) [reply dotnet/src/dotnetcore/Providers/Messaging/GXAzureEventGrid/AzureEventGrid.cs]: ready.

File: dotnet/src/dotnetcore/Providers/Messaging/GXAzureEventGrid/AzureEventGrid.cs [hunk 4]
```diff
@@ -0,0 +1,327 @@
...
+		private async Task<bool> sendEvtAsync(CloudEvent cloudEvent)
+		{
+			try
+			{
+				await _client.SendEventAsync(cloudEvent).ConfigureAwait(false);
+				return true;
+			}
+			catch (Exception ex)
+			{
+				throw ex;
```
Comment (ggallotti) [inline dotnet/src/dotnetcore/Providers/Messaging/GXAzureEventGrid/AzureEventGrid.cs:213]: The try-catch block is not really needed here.

File: dotnet/src/dotnetcore/Providers/Messaging/GXAzureEventGrid/AzureEventGrid.cs [hunk 5]
```diff
@@ -0,0 +1,327 @@
...
+		private async Task<bool> sendEvtsAsync(IEnumerable<CloudEvent> cloudEvents)
+		{
+			try
+			{
+				await _client.SendEventsAsync(cloudEvents).ConfigureAwait(false);
+				return true;
+			}
+			catch (Exception ex)
+			{
+				throw ex;
```
Comment (ggallotti) [inline dotnet/src/dotnetcore/Providers/Messaging/GXAzureEventGrid/AzureEventGrid.cs:230]: The try-catch block is not really needed here.

File: dotnet/src/dotnetcore/Providers/Messaging/GXAzureEventGrid/AzureEventGrid.cs [hunk 6]
```diff
@@ -0,0 +1,327 @@
...
+		private async Task<bool> sendEventGridSchemaEventsAsync(IEnumerable<EventGridEvent> evts)
+		{
+			try
+			{
+				await _client.SendEventsAsync(evts).ConfigureAwait(false);
+				return true;
+			}
+			catch (Exception ex)
+			{
+				throw ex;
```
Comment (ggallotti) [inline dotnet/src/dotnetcore/Providers/Messaging/GXAzureEventGrid/AzureEventGrid.cs:196]: The try-catch block is not really needed here.

File: dotnet/src/dotnetcore/Providers/Messaging/GXAzureEventGrid/AzureEventGrid.cs [hunk 7]
```diff
@@ -0,0 +1,327 @@
...
+		private async Task<bool> sendEventGridSchemaEventAsync(EventGridEvent evt)
+		{
+			try
+			{
+				await _client.SendEventAsync(evt).ConfigureAwait(false);
+				return true;
+			}
+			catch (Exception ex)
+			{
+				throw ex;
```
[the remark this answers is not in the document]
Comment (sjuarezgx) [reply dotnet/src/dotnetcore/Providers/Messaging/GXAzureEventGrid/AzureEventGrid.cs]: All try catch blocks removed. The test is automated in fullgx.

**Review by ggallotti — APPROVED** [review]

**Pull request merged**